### DISCUSSION TITLE
Is it possible to impose a document

### DISCUSSION THREAD

**Discussion by johnmapeson** [pdfcpu/pdfcpu#928] [Q&A] [answered]
Hello again, Horst! :-)

Could you show how to make `nup` to put 2 pages side by side on the left and right sides of a paper? I tried `pdfcpu nup -- output.pdf 2 input.pdf`, but it puts the 2nd page below the 1st instead of on the right side. If I understood documentation correclty, I need to use Landscape paper orientation, but I cannot figure out how to do it.

**Comment by hhrutter** [comment]
There is `orientation` and `enforce` to play around with.

**Comment by johnmapeson** [comment]
Weird. There is probably a bug, or several.

- `pdfcpu nup -- "orientation:<...>"  output.pdf 2 input.pdf`: No matter which orienatation I choose (`rd`, `dr`, ...), the output file looks the same as if I don't use `orientation` at all.

- `pdfcpu nup -- "orientation:<...>, enforce:true"  output.pdf 2 input.pdf`: No matter which orienatation I choose, there is the `ambiguous parameter prefix "enforce"` error.

**Comment by johnmapeson** [comment]
I have found a solution based on using `grid`, but I suppose a solution based on using `nup` would be more appropriate for this kind of work.

```
pdfcpu grid -- output.pdf 1 2 input.pdf
```

**Comment by hhrutter** [comment]
I have to apologize - bugs happen.
This is open source as in people who dedicate their spare time on top of a full time job and family life.
I have around 80 open issues and new features are being worked on as well.

This is a discussion forum.
If you think you have discovered a bug, please open an issue for each one.

If you are having fun and want to see your issues adressed with priority consider becoming a [pdfcpu sponsor](https://github.com/sponsors/hhrutter) 💚

**Comment by johnmapeson** [comment]
Probably there are no bugs here actually. When you said "There is `orientation` and `enforce` to play around with", I thought you meant it *must* work for my particular case. But in fact what you said then might have been just a guess, and then everything probably works correctly.

**Comment by hhrutter** [comment, marked as answer]
The docs at pdfcpu.io need improvement.

There maybe a confusion around `form` vs. `orientation`, I'll try to explain:

`form` is the size of the page onto which the individual `nup` and `grid` artifacts get rendered.
You may append `L` or `P` respectively to the form name in order to configure landscape or portrait mode.
Eg. A4L configures the destination page/form size as A4 landscape.

Then when individual artifacts get rendered in row/column fashion onto the destination page `orientation` defines the rendering 
run down mode: right down, down right, left down, down left.

This still does not define the orientation of the individual artifacts  that get rendered.
The default is a best-fit approach, which factors in the aspect ratios and portrait/landscape mode of the destination page
together with the aspect ratios of the individual artifacts in order to get the `best` result.

The new `enforce` flag toggles uasge of this (best-fit) approach.
`enforce` for `nup` and `grid` is not yet documented at pdfcpu.io so that's missing
but `pdfcpu help nup` and `pdfcpu help grid` explain `enforce`.
